pkg/forklift-api/webhooks/validating-webhook/admitters/storage-admitter.go

```diff
@@ -0,0 +1,125 @@
+package admitters
+
+import (
+	"context"
+	v1 "k8s.io/api/storage/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"encoding/json"
+	"fmt"
+	admissionv1 "k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"net/http"
+
+	"github.com/konveyor/forklift-controller/pkg/apis"
+	api "github.com/konveyor/forklift-controller/pkg/apis/forklift/v1beta1"
+	"github.com/konveyor/forklift-controller/pkg/forklift-api/webhooks/util"
+)
+
+type StorageAdmitter struct {
+}
+
+func (admitter *StorageAdmitter) Admit(ar *admissionv1.AdmissionReview) *admissionv1.AdmissionResponse {
+	log.Info("storage admitter was called")
+	raw := ar.Request.Object.Raw
+
+	if len(raw) == 0 {
+		// plan is being deleted, passing
+		return &admissionv1.AdmissionResponse{
+			Allowed: true,
+		}
+	}
```

Comment:
if no validation is required when deleting a plan (which makes sense), we can define the webhook in a way that it's not involved in deletion - see [this](https://github.com/kubev2v/forklift/blob/main/operator/roles/forkliftcontroller/templates/api/validatingwebhookconfiguration-forklift-api.yml.j2#L35-L37)

Reply:
Missed that one! I did saw the new matcher which requires enabling featuregate(alpha) so i went with this workaround. Your suggestion is much better, thanks.